File: 03.LongestSubstringWithoutRepeatingCharacters/main.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        length = len(s)
        max_length = 0

        for i in range(length):
            arr = []
            for j in range(i, length):
                if s[j] in arr:
                    break
                else:
                    arr.append(s[j])
            if len(arr) > max_length:
                max_length = len(arr)

        return max_length


solution = Solution()

print(solution.lengthOfLongestSubstring("abcabcbb"))
print(solution.lengthOfLongestSubstring("bbbbb"))
print(solution.lengthOfLongestSubstring("pwwkew"))

File: 03.LongestSubstringWithoutRepeatingCharacters/main.rb
class Solution
  def length_of_longest_substring(s)
    length = s.length
    max_length = 0

    (0...length).each do |i|
      arr = []
      (i...length).each do |j|
        if arr.include?(s[j])
          break
        else
          arr << s[j]
        end
      end
      max_length = [max_length, arr.size].max
    end

    max_length
  end
end

solution = Solution.new

puts solution.length_of_longest_substring("abcabcbb")
puts solution.length_of_longest_substring("bbbbb")
puts solution.length_of_longest_substring("pwwkew")
